flink-runtime/src/main/scala/org/apache/flink/runtime/taskmanager/TaskManager.scala

```diff
@@ -411,6 +411,23 @@ class TaskManager(
             log.debug(s"Cannot find task to fail for execution ${executionID})")
           }
 
+        // stops a task
+        case StopTask(executionID) =>
+          val task = runningTasks.get(executionID)
+          if (task != null) {
+            try {
+              task.stopExecution()
+              sender ! new TaskOperationResult(executionID, true)
+            } catch {
+              case t: Throwable =>
+                        sender ! new TaskOperationResult(executionID, false, t.getMessage())
```

Comment:
If this can be a user code exception, then we have to serialize it and deserialize it with the correct class loader on the receiving side. Otherwise akka will fail.